src/ssl.c

```diff
@@ -49797,6 +50464,56 @@ int wolfSSL_BN_mod_add(WOLFSSL_BIGNUM *r, const WOLFSSL_BIGNUM *a,
 
 #ifdef WOLFSSL_KEY_GEN
 
+int wolfSSL_BN_generate_prime_ex(WOLFSSL_BIGNUM* prime, int bits,
+    int safe, const WOLFSSL_BIGNUM* add, const WOLFSSL_BIGNUM* rem,
+    WOLFSSL_BN_GENCB* cb)
+{
+    int ret = WOLFSSL_SUCCESS;
+#ifdef WOLFSSL_SMALL_STACK
+    WC_RNG* rng = NULL;
+#else
+    WC_RNG rng[1];
+#endif
+
+    (void)cb;
+
+    WOLFSSL_ENTER("wolfSSL_BN_generate_prime_ex");
+
+    if (safe == 1 || add != NULL || rem != NULL) {
+        /* These parameters aren't supported, yet. */
+        ret = WOLFSSL_FAILURE;
+    }
+
+#ifdef WOLFSSL_SMALL_STACK
+    if (ret == WOLFSSL_SUCCESS) {
+        rng = (WC_RNG*)XMALLOC(sizeof(WC_RNG), NULL, DYNAMIC_TYPE_RNG);
+        if (rng == NULL) {
+            ret = WOLFSSL_FAILURE;
+        }
+    }
+#endif
+    if (ret == WOLFSSL_SUCCESS && wc_InitRng(rng) != 0) {
+        ret = WOLFSSL_FAILURE;
+    }
+
+    if (ret == WOLFSSL_SUCCESS) {
+        if (mp_rand_prime((mp_int*)prime->internal, (bits + 7) / 8, rng, NULL)
+                != MP_OKAY) {
+            ret = WOLFSSL_FAILURE;
+        }
+    }
+
+    wc_FreeRng(rng);
```

Comment:
On a failure to wc_InitRng this will have unpredictable behavior. Either add flag to indicate cleanup is needed or memset the RNG struct.